client/server/pages/index.js

```diff
@@ -586,7 +587,7 @@ export default function pages() {
 		res.redirect( redirectUrl );
 	} );
 
-	if ( process.env.NODE_ENV !== 'development' ) {
+	if ( ! isJetpackCloud() && process.env.NODE_ENV !== 'development' ) {
```

Comment:
Routing defined in this condition doesn't apply to cloud.

Reply:
I think that all the special server-side route handlers, starting with `/`, continuing through `/sites/:site/:section`, and ending with `/menus`, `/domains`, `/browsehappy` and `/support-user`, are specific to Calypso Blue and should be guarded with the `if ( ! isJetpackCloud() )` condition. And ideally extracted to a separate function.

The only things that are universal across Blue and Green are the `app.use` middlewares and the `sections.forEach( registerSectionRoutes )` loop.

I also think that the `development` condition is redundant: these handlers can be registered in dev mode, too.

Reply:
Thanks for the recommendations, @jsnajdr.